hs_core/models.py

```diff
@@ -1940,14 +1941,39 @@ def create(cls, resource, file, folder=None, source=None, move=False):
 
         else:  # if file is not an open file, then it's a basename (string)
             if file is None and source is not None:
-                root, file = os.path.split(source)  # take from source path
-            target = get_resource_file_path(resource, file, folder=folder)
-            if source is not None:
+                if __debug__:
+                    assert(isinstance(source, basestring))
+                # source is a path to an iRODS file to be copied here.
+                root, newfile = os.path.split(source)  # take file from source path
+                # newfile is where it should be copied to.
+                target = get_resource_file_path(resource, newfile, folder=folder)
                 istorage = resource.get_irods_storage()
+                if __debug__:
+                    if not istorage.exists(source):
+                        raise ValidationError("ResourceFile.create: source {} of copy not found"
+                                              .format(source))
                 if not move:
                     istorage.copyFiles(source, target)
                 else:
                     istorage.moveFile(source, target)
+                if __debug__:
+                    if not istorage.exists(target):
+                        raise ValidationError("ResourceFile.create: copy to target {} failed"
+                                              .format(target))
```

Comment:
@alvacouch Now that I think about it, agreed for this specific post-operation validation check, enclosing validations with a debug directive is better since if it fails, iRODS will raise an exception. On second thought, we don't need these post-operation validation checks at all since iRODS will raise an exception when any operation of copy or move fails. Calling istorage.exists() does introduce another subprocess call, so it will slow down things a bit depending on the usage. I'd suggest to remove these post-copy/move-operation validation checks, but defer to your judgement on whatever works best for now.